src/main/java/net/pms/dlna/LibMediaInfoParser.java

```diff
@@ -907,4 +947,15 @@ private static double getDuration(String value) {
 
 		return (h * 3600) + (m * 60) + s;
 	}
+
+	//TODO: (Nad) Ask valib about reasoning, shouldn't be used for misspelled titles
+	private static String correctLanguage(String language) {
+		if ("Chinese AAC".equals(language)) {
+			return "Chinese";
+		}
+		if ("Hebrwe".equals(language)) {
+			return "Hebrew";
+		}
+		return language;
+	}
```

Comment:
@valib I moved your "dirty fixes" from Iso639 and put it here - adding it as separate languages as you did would cause all kind of bugs. But, the more I look at this the less I understand: How is this relevant to do, I mean try to parse the languages from the "*title*", and if so "compensate" for wrong spellings in this way?

I suspect that the misspellings listed here is only some random misspellings that has been in some of your test files. Making code with hacks for your test files only is pretty useless IMO. Given that it's common to find the language name in the "tite" field, a different search must be done AFAICU. You are matching the whole "title" to the whole language name, but the title is likely to also contain other information (like AAC). What must be done, is to search the title for valid language codes. It can be tricky though, as you can get false positives and it would probably also take a bit of time since you have to search the string for all possible language names.

Correcting for possible spelling mistakes like "Hebrwe" is pretty pointless IMO, just try to imagine how many possible spelling mistakes there are only for the English language names for all languages. Then, add the fact that it might not be written in English at all....

Reply:
@Nadahar this "dirty fix" I made on the user request. I know that it can be done for all misspelled languages but in that particular case I checked the internet and it seems that the "Hebrwe" is quite common misspelling to be worth to be fixed. On the other hand we can simply say "please users don't borther us with the problems which are on your site" but this "dirty fix" is simple and one of the special fixes like for PS3, Xbox and Sony renderers we already made. The question is what is better. To fix it or refuse it?

Reply:
@valib That's why I wanted to ask. I suspect there are many common misspellings, so it feels a bit funny to only have the one for Hebrew. The "Chinese" fix must be done another way as I said, you can't have one version for every language with every possible combination with other words that might be in the title.

Reply:
@valib I've refactored most of this into ```Iso639``` in 15f8b4535795a16cc19c5b5c149c388bc8c3c1e7. As you can see I've added common misspellings for many other language names as well.

Reply:
@Nadahar there is something wrong. When I test this latest update subtitles are not added to be transcoded or streamed with video. They are only shown in the TRANSCODE folder.

See the [ums_dbg.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/1604816/ums_dbg.zip)

Reply:
@valib I'm not finished with the fixes for the first commit, so that's only natural. I'm trying to "clean up"/reduce the other commits to make it easier to handle the rewrite, which is why I concentrated on the "mediainfo" commit right now. So, it's not possible to test this yet - but the ```Iso639``` code is supposed to be finished, so you can review that if you want to.